src/olympia/shelves/admin.py

```diff
@@ -1,6 +1,45 @@
-from django.contrib import admin
+import requests
+
+from rest_framework.reverse import reverse as drf_reverse
+
+from django.contrib import admin, messages
+
+from .forms import ShelfForm
 
 
 class ShelfAdmin(admin.ModelAdmin):
     list_display = ('title', 'shelf_type')
     actions = ['delete_selected']
+    form = ShelfForm
+
+    def message_user(self, request, message, level=messages.INFO,
+                     extra_tags='', fail_silently=False):
+        pass
+
+    def save_model(self, request, obj, form, change):
+        baseUrl = "https://addons.mozilla.org"
+
+        if obj.shelf_type in ('extension', 'search', 'theme'):
+            api = drf_reverse('v4:addon-search')
+        elif obj.shelf_type == 'categories':
+            api = drf_reverse('v4:category-list')
+        elif obj.shelf_type == 'collections':
+            api = drf_reverse('v4:collection-list')
+        elif obj.shelf_type == 'recommendations':
+            api = drf_reverse('v4:addon-recommendations')
+
+        url = baseUrl + api + obj.criteria.lower()
+        response = requests.get(url)
+        results = response.json()
```

Comment:
this is all duplicated with the code in forms, and would mean two identical requests, so we don't want that.  I'm guessing this is just because you need the response to include the count in the messages.  

If so, lets just hack around that - store the response on the form instance `self.response = response.json()`, then add it to the Shelf instance (in the form's save method), then it'll be available as `obj.response` in `save_model`.